library/swift/test/HttpBridgeTests/ReceiveDataTest.swift

```diff
@@ -0,0 +1,78 @@
+import Envoy
+import EnvoyEngine
+import Foundation
+import XCTest
+
+final class ReceiveDataTests: XCTestCase {
+  func testReceiveData() throws {
+    // swiftlint:disable:next line_length
+    let apiListenerType = "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager"
+    // swiftlint:disable:next line_length
+    let assertionFilterType = "type.googleapis.com/envoymobile.extensions.filters.http.assertion.Assertion"
+    let config =
+    """
+    static_resources:
+      listeners:
+      - name: base_api_listener
+        address:
+          socket_address:
+            protocol: TCP
+            address: 0.0.0.0
+            port_value: 10000
+        api_listener:
+          api_listener:
+            "@type": \(apiListenerType)
+            stat_prefix: hcm
+            route_config:
+              name: api_router
+              virtual_hosts:
+                - name: api
+                  domains:
+                    - "*"
+                  routes:
+                    - match:
+                        prefix: "/"
+                      direct_response:
+                        status: 200
+                        body:
+                          inline_string: response_body
+            http_filters:
+              - name: envoy.filters.http.assertion
+                typed_config:
+                  "@type": \(assertionFilterType)
+                  match_config:
+                    http_request_headers_match:
+                      headers:
+                        - name: ":authority"
+                          exact_match: example.com
+              - name: envoy.router
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+    """
+    let expectation = self.expectation(description: "Run called with expected data")
+    let client = try EngineBuilder(yaml: config)
+      .addLogLevel(.debug)
+      .addFilter(factory: DemoFilter.init)
+      .build()
+      .streamClient()
+
+    let requestHeaders = RequestHeadersBuilder(method: .get, scheme: "https",
+                                               authority: "example.com", path: "/test")
+      .addUpstreamHttpProtocol(.http2)
+      .build()
+    client
+      .newStreamPrototype()
+      .setOnResponseHeaders { responseHeaders, _ in
+         XCTAssertEqual(200, responseHeaders.httpStatus)
```

Comment:
We should add a second expectation here and wait for it below as well - there's no validation as-is that this assert is actually called

Reply:
How come there isn't? If the expectation below doesn't fulfill then the test will fail. Meaning that if the test finishes, this code must have ran?

Reply:
My concern is that it relies on the (hopefully always valid) assumption that `onResponseData` will always be called after `onResponseHeaders` (since the expectation being fulfilled is fulfilled upon receiving response data, but nothing is explicitly fulfilled on headers). Adding an expectation here would prevent the theoretical issue of the filter responding with data callbacks but no headers